### PR TITLE
Add x264 library package

### PR DESCRIPTION
**Pull request by ppodhajski**

### CODE REVIEW

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
@pramodk could you please have a look why this jobs fails?

**Comment by matz-e** [comment]
```
libavcodec/libx264.c: In function 'X264_frame':
libavcodec/libx264.c:282:9: error: 'x264_bit_depth' undeclared (first use in this function)
if (x264_bit_depth > 8)
^~~~~~~~~~~~~~
libavcodec/libx264.c:282:9: note: each undeclared identifier is reported only once for each function it appears in
libavcodec/libx264.c: In function 'X264_init_static':
libavcodec/libx264.c:892:9: error: 'x264_bit_depth' undeclared (first use in this function)
if (x264_bit_depth == 8)
^~~~~~~~~~~~~~
```
Weird…

**Comment by ppodhajski** [comment]
retest this please

File: var/spack/repos/builtin/packages/ffmpeg/package.py [hunk 1]
```diff
@@ -13,12 +13,18 @@ class Ffmpeg(AutotoolsPackage):
...
+    version('3.2.4', 'd3ebaacfa36c6e8145373785824265b4')
+    version('4.2', '306bde5f411e9ee04352d1d3de41bd36')
 
     variant('shared', default=True,
             description='build shared libraries')
+    variant('x264', default=True,
+            description='enable x264 encoder')
 
     depends_on('yasm@1.2.0:')
+    depends_on('x264', when='+x264')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/ffmpeg/package.py]: Maybe try this? 
```suggestion
    depends_on('x264', when='@4.2: +x264')
```

**Comment by matz-e** [comment]
NB: all my notifications end up with no visible changes due to the force-pushes… would be nice to see what actually changed, especially since we just squash at the very end anyways

FWIW, `deploy/packages/tools.yaml` builds `ffmpeg@3.2.4`. That should probably changed by removing the version pin.

**Comment by ppodhajski** [comment]
ok, can do it by introducing a new commit but you can still see history and changes of force pushes, right?

**Comment by matz-e** [comment]
It's just that I get a notification about changes, but the links open a page with no changes. With new commits, that doesn't happen.

File: deploy/packages/tools.yaml [hunk 2]
```diff
@@ -25,7 +25,6 @@ packages:
       - doxygen@1.8.15
       - emacs@26.1
       - environment-modules@3.2.10
```
Comment (matz-e) [inline deploy/packages/tools.yaml:27]:
```suggestion
       - environment-modules@3.2.10
       - ffmpeg
```

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
@matz-e: strange

> 18:44:03  ==> Error: Package emacs has no variant set(['environment-modules', 'ffmpeg'])!

**Review by matz-e — APPROVED** [review]

**Pull request merged**